ethexe/signer/src/signer.rs

```diff
@@ -0,0 +1,201 @@
+};
+
+use anyhow::Result;
+use std::{
+    fs,
+    path::PathBuf,
+    sync::{Arc, RwLock, RwLockReadGuard, RwLockWriteGuard},
+};
+
+/// Signer which signs data using owned key store.
+#[derive(Debug, Clone)]
+pub struct Signer {
+    key_store: Arc<RwLock<dyn KeyStorage>>,
+}
+
+impl Signer {
+    /// Create a new signer with a key store.
+    pub fn new(key_store: impl KeyStorage) -> Self {
+        Self {
+            key_store: Arc::new(RwLock::new(key_store)),
+        }
+    }
+
+    /// Create a new signer with a key store location.
+    pub fn fs(path: PathBuf) -> Self {
+        fs::create_dir_all(path.as_path()).expect("Cannot create key store dir");
+
+        Self {
+            key_store: Arc::new(RwLock::new(FSKeyStorage {
+                path,
+                _tmp_dir: None,
+            })),
+        }
+    }
+
+    /// Create a new signer with a key temporary key store location.
+    pub fn fs_temporary() -> Self {
+        Self::new(FSKeyStorage::temporary())
+    }
+
+    /// Create a new signer with an in-memory key store.
+    pub fn memory() -> Self {
+        Self {
+            key_store: Arc::new(RwLock::new(MemoryKeyStorage::default())),
+        }
+    }
+
+    /// Create wrapped signer for contract address.
+    pub fn contract_signer(&self, contract_address: Address) -> ContractSigner {
+        ContractSigner::new(self.clone(), contract_address)
+    }
+
+    /// Create a ECDSA recoverable signature with `Electrum` notation for the `v` value.
+    ///
+    /// For more info about `v` value read [`RawSignature`] docs.
+    pub fn raw_sign_digest(&self, public_key: PublicKey, digest: Digest) -> Result<RawSignature> {
+        let private_key = self.storage().get_private_key(public_key)?;
+
+        RawSignature::create_for_digest(private_key, digest)
+    }
+
+    // TODO #4365
+    /// Create a ECDSA recoverable signature.
+    pub fn sign<T>(&self, public_key: PublicKey, data: T) -> Result<Signature>
+    where
+        Digest: From<T>,
+    {
+        let digest = Digest::from(data);
+        let private_key = self.storage().get_private_key(public_key)?;
+
+        Signature::create_for_digest(private_key, digest)
+    }
+
+    /// Create a ECDSA recoverable signature packed with data together.
+    pub fn signed_data<T: Sized>(&self, public_key: PublicKey, data: T) -> Result<SignedData<T>>
+    where
+        for<'a> Digest: From<&'a T>,
+    {
+        self.sign(public_key, &data)
+            .map(|signature| SignedData::new(data, signature))
+    }
+
+    /// Generate a new private key and return a public key for it.
+    pub fn generate_key(&self) -> Result<PublicKey> {
+        let (secp256k1_secret_key, secp256k1_public_key) =
+            secp256k1::generate_keypair(&mut secp256k1::rand::thread_rng());
+
+        let public_key: PublicKey = secp256k1_public_key.into();
+        let private_key: PrivateKey = secp256k1_secret_key.into();
+
+        log::debug!("New secret key generated: {private_key}");
```

Comment:
remove this debug please and put necessary one in `cli/src/commands/key.rs/KeyCommand::Generate` handling

Reply:
This debug is not only for cli - it is also for tests

Reply:
you can put debug in test env than

Reply:
it's super unsafe to log private keys

Reply:
The main reason of generate is to generate new keys and show them to use if he wants.
If you wanna not to print it
- do not use RUST_LOG=debug
- or RUST_LOG=ethexe_signer=off

Reply:
removed